framework/include/utils/RankFourTensorImplementation.h

```diff
@@ -796,10 +797,17 @@ RankFourTensorTempl<T>::fillSymmetricIsotropicFromInputVector(const std::vector<
 
 template <typename T>
 void
-RankFourTensorTempl<T>::fillSymmetricIsotropic(T lambda, T G)
+RankFourTensorTempl<T>::fillSymmetricIsotropic(const T & lambda, const T & G)
 {
-  fillSymmetric9FromInputVector(
-      {lambda + 2.0 * G, lambda, lambda, lambda + 2.0 * G, lambda, lambda + 2.0 * G, G, G, G});
+  // clang-format off
+  fillSymmetric21FromInputVector(std::array<T, 21>
```

Comment:
Try removing array type and make it single braces. That should use std:: initializer_list as the type. Will look a bit nicer. Didn't think of that yesterday.

Reply:
Is it worth precomputing `lambda + 2.0 * G`? maybe not

Reply:
the compiler almost certainly will make this simple optimization on its own

Reply:
@dschwen How will it know that I want to call filleSymmetric21FromInputVector with  an array and not a vector if I don't explicitly construct the array?

Reply:
It will call it with neither std::vector nor std::array, but with std::initializer_list

Reply:
@dschwen I want to call it with an array so that the memory is allocated on the heap.  I didn't want to reconstruct a vector on the stack every time I call this.  I don't think the initilizer_list would be on the heap.  I thought this would be cheaper for a list of ADReals.  I did run some timings with Gary's model and std::vector is the same as std::array so it doesn't really matter.

Reply:
?!

Reply:
OK!  I fixed my comment but I'll make the change.

Reply:
I can't use the initilizer_list ```std::initializer_list<double>' does not provide a subscript operator```   I need random access into the array to fill the rank for tensor.